inc/namespace.php

```diff
@@ -1,9 +1,115 @@
 <?php
+/**
+ * Core platform functions.
+ *
+ * @package hm-platform
+ */
 
 namespace HM\Platform;
 
 use Aws\Sdk;
 
+/**
+ * Retrieve the configuration for HM Platform.
+ *
+ * The configuration is defined by merging the defaults set by modules
+ * with any overrides present in composer.json.
+ *
+ * @return array Configuration data.
+ */
+function get_config() : array {
+	static $config = [];
+
+	if ( empty( $config ) ) {
+		$config = get_merged_config();
+	}
+
+	/**
+	 * Filter the entire platform config.
+	 *
+	 * @param array $config The full config array.
+	 */
+	$config = apply_filters( 'hm-platform.config', $config );
```

Comment:
This is inconsistent on repeated calls, as the second time around it will re-filter the previously-filtered code, so might break. Imagine something like:

```
add_filter( 'hm-platform.config', function ( $data ) {
    $data['foo'] .= 'blah';
    return $data;
} );
```